src/VisualStudio/Core/Def/SampleToolWindow/DocumentOutlineHelper.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServices
+{
+    internal static class DocumentOutlineHelper
+    {
+        internal static ObservableCollection<DocumentSymbolViewModel> GetDocumentSymbols(DocumentSymbol[]? documentSymbols)
+        {
+            var documentSymbolModels = new ObservableCollection<DocumentSymbolViewModel>();
+            if (documentSymbols is null || documentSymbols.Length == 0)
+            {
+                return documentSymbolModels;
+            }
+
+            foreach (var documentSymbol in documentSymbols)
+            {
+                var documentSymbolModel = new DocumentSymbolViewModel(documentSymbol);
+                documentSymbolModel.Children = GetDocumentSymbols(documentSymbol.Children);
+                documentSymbolModels.Add(documentSymbolModel);
+            }
+
+            return Sort(documentSymbolModels, SortOption.Order);
+        }
+
+        internal static ObservableCollection<DocumentSymbolViewModel> Sort(ObservableCollection<DocumentSymbolViewModel> documentSymbolModels, SortOption sortOption)
+        {
+            if (documentSymbolModels.Count <= 1)
```

Comment:
Doesn't this mean that we never sort items if there is only one item at the top of the file? wouldn't we want to bail out of our recursive function once there are no more children instead of just when there is only one or less items?

Reply:
You are correct, this doesn't do what I wanted it to haha